### ISSUE TITLE
Choosing a frontend framework - React or VueJS

### ISSUE THREAD

**Issue by codesword** [codesword/entranic-architecture#1]
React and Vue share many similarities. They both:
- utilize a virtual DOM
- provide reactive and composable view components
- maintain focus in the core library, with concerns such as routing and global state management handled by companion libraries.

Differences btw React and VueJS
**Jsx vs Templates**
React uses jsx templates eg 
```jsx
render () {
  let { items } = this.props
  let children
  if (items.length > 0) {
    children = (
      <ul>
        {items.map(item =>
          <li key={item.id}>{item.name}</li>
        )}
      </ul>
    )
  } else {
    children = <p>No items found.</p>
  }
  return (
    <div className='list-container'>
      {children}
    </div>
  )
}
```
Jsx has a number of advantages:
- You can use the power of a full programming language (JavaScript) to build your view.
- The tooling support (e.g. linting, type checking, editor autocompletion) for JSX is in some ways more advanced than what’s currently available for Vue templates.

VueJS uses html style templates.
```html
<template>
  <div class="list-container">
    <ul v-if="items.length">
      <li v-for="item in items">
        {{ item.name }}
      </li>
    </ul>
    <p v-else>No items found.</p>
  </div>
</template>
```
The above html style templates gives a number of advantages:
- Fewer implementation and stylistic decisions have to be made while writing a template
- A template will always be declarative
- Any valid HTML is valid in a template
- It reads more like English (e.g. for each item in items)
- Advanced versions of JavaScript are not required to increase readability
Which is easier to reason about and be used with html preprocessors eg jade(pug).
- View supports render functions and Jsx if needed.
- You can use html preprocessors(eg jade) to author vue templates.

**Learning Curve**
VueJS is more intuitive and has an easier learning curve unlike React which is more abstract with a steep learning curve.

**Popularity**
React is very popular and is backed by tech giant facebook while on the other hand, vueJS has rising popularity.

Check out [this link](https://vuejs.org/v2/guide/comparison.html#React) for more indepth comparison.

**Comment by codesword** [comment]
I will like to hear your view on the framework we should choose.

**Comment by thechinedu** [comment]
I don't think we should spend too much time on this, for the most parts, both frameworks/libraries will give the same or very similar performance benefits in terms of speed and ease of use. 

However, if we are all new to both technologies (haven't used either extensively), then we should just go with `VueJs` as you have said it is more beginner friendly.

**Comment by 0sc** [comment]
I would vote for ReactJS only because it's popular and has lots of resources from an engaging community. I'm currently more excited about VueJS and would gladly love to try it out but I don't think the projected timeline for this project allows enough room for avoidable experiment.

My rule of thumb, for now, is _where quality wouldn't be affected go with the tool the team has greater familiarity_.

**Comment by Adebolavictor** [comment]
If most of us are fairly new to these technologies like I am then I would suggest we go for VueJs, otherwise ReactJs would be great cos of the community support and company behind it

**Comment by codesword** [comment]
@andela-cdaniel @andela-ooranagwa @Adebolavictor Thanks for responding. After doing more digging. I discovered that VueJS also has a lot of resources. Probably not as much as React though. I will like to stick with VueJS because of the beginner friendliness. Before the project kicks off, I will try to create a starter template that has most of the things discussed in [frontend architecture](https://github.com/codesword/entranic-architecture). This way, the frontend team can focus on implementation.
